src/entity.ts

```diff
@@ -383,49 +384,101 @@ export class Entity {
 	}
 }
 
+export type EntityPropertiesOfType<T> = {
+    [P in keyof T]:
+		T[P] extends (infer TItem)[]
+			? ObservableArray<(
+				TItem extends string ? string :
+				TItem extends number ? number :
+				TItem extends boolean ? boolean :
+				TItem extends Date ? Date :
+				EntityOfType<TItem>
+			)>
+			: (
+				T[P] extends string ? string :
+				T[P] extends number ? number :
+				T[P] extends boolean ? boolean :
+				T[P] extends Date ? Date :
+				EntityOfType<T[P]>
+			) | null;
+};
+
+type EntityOfTypeMemberOverrides<T> = {
+    update(args: EntityArgsOfType<T>): Promise<void>;
+};
+
+export type EntityOfType<T> = EntityPropertiesOfType<T> & EntityOfTypeMemberOverrides<T> & Entity; // Omit<Entity, keyof EntityOfTypeMemberOverrides<T>> & ;
+
+export type TEntityConstructor<T> = {
+    new(id: string, args?: EntityArgsOfType<T>): EntityOfType<T>;
+    new(args?: EntityArgsOfType<T>): EntityOfType<T>;
+	meta: TypeOfType<T>;
+};
+
+export type EntityArgsOfType<T> = Partial<{
```

Comment:
_info_: Define type `EntityArgsOfType<T>`, where `T` is the "schema" of the type (the properties it has and their TS types). This is intended to be used for constructors, `create` calls, `update` calls, etc. All properties are optional, so they can be omitted. Non-lists can be null, but lists should not be null. The supported value types are covered, and then anything else is assumed to be an entity property. Entity property values can be an instance of the entity or the supported "args" (i.e. this type) for that entity type.

Reply:
```ts
type Example = {
  text: string;
  values: number[];
  ref: OtherEntity;
  children: ChildEntity[];
};

type ExampleEntityArgs = EntityArgsOfType<Example> = {
  text?: string | null;
  values?: number[];
  ref?: EntityOfType<OtherEntity> | EntityArgsOfType<OtherEntity> | null;
  children?: (EntityOfType<ChildEntity> | EntityArgsOfType<ChildEntity>)[];
};
```

Reply:
Also, `readonly` properties are excluded, since they would represent calculated properties that can't be set.